### PR TITLE
Add 1.14 release blog post.

### PR DESCRIPTION
**Pull request by steveklabnik**
r? @rust-lang/core

### CODE REVIEW

File: _posts/2016-12-22-Rust-1.14.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+$ rustup target add wasm32-unknown-emscripten
+$ echo 'fn main() { println!("Hello, Emscripten!"); }' > hello.rs
+$ rustc --target=wasm32-unknown-emscripten hello.rs
+$ node hello.js
+```
+
+[wasm]: https://users.rust-lang.org/t/compiling-to-the-web-with-rust-and-emscripten/7627
+[emscripten]: http://kripken.github.io/emscripten-site/docs/getting_started/downloads.html
+
+The community has been doing intertesting, experimental work in this area: see
```
Comment (durka) [inline _posts/2016-12-22-Rust-1.14.md]: interesting

File: _posts/2016-12-22-Rust-1.14.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+As always, you can [install Rust 1.14.0][install] from the appropriate page on our
+website, and check out the [detailed release notes for 1.14.0][notes] on GitHub.
+1230 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1140-2016-12-22
+
+### What's in 1.14.0 stable
+
+One of the biggest features in Rust 1.14 isn't actually in Rust 1.14: the
```
Comment (durka) [inline _posts/2016-12-22-Rust-1.14.md]: perhaps "isn't in the compiler"?

File: _posts/2016-12-22-Rust-1.14.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+```rust
+let p = Point(0, 1, 2);
+
+match p {
+    Point(x, ..) => println!("x is {}", x),
+}
+```
+
+This was a inconsistency, and so with RFC 1492 stabilized, compiles fine as of
```
Comment (nagisa) [inline _posts/2016-12-22-Rust-1.14.md]: a inconsistency

File: _posts/2016-12-22-Rust-1.14.md (hunk 2 above)
Comment (nagisa) [inline _posts/2016-12-22-Rust-1.14.md]: reads weird

File: _posts/2016-12-22-Rust-1.14.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+#### Library stabilizations
+
+There have been a number of additions to the standard library, but they don't
+fit into particularly nice categories for this release. Here's the highlights:
+
+* [`println!()`, with no arguments, prints newline][36825].
+  Previously, an empty string was required to achieve the same.
+* [`Wrapping` impls standard binary and unary operators, as well as
+   the `Sum` and `Product` iterators][37356]
```
Comment (durka) [inline _posts/2016-12-22-Rust-1.14.md]: actually the operator impls were already there, this just adds them for references

File: _posts/2016-12-22-Rust-1.14.md (hunk 2 above)
Comment (Mark-Simulacrum) [inline _posts/2016-12-22-Rust-1.14.md]: Possible change: "One of the biggest features in Rust 1.14 isn't actually in the language itself:"?

File: _posts/2016-12-22-Rust-1.14.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+* [Implement `From<Cow<str>> for String` and `From<Cow<[T]>> for
+  Vec<T>`][37326]
+* [Expand `.zip()` specialization to `.map()` and `.cloned()`][37230]
+* [Implement `RefUnwindSafe` for atomic types][37178]
+* [Specialize `Vec::extend` to `Vec::extend_from_slice`][37094]
+* [Don't reuse `HashMap` random seeds][37470]
```
Comment (durka) [inline _posts/2016-12-22-Rust-1.14.md]: maybe mention why? (mitigates one type of HashDoS attack)

File: _posts/2016-12-22-Rust-1.14.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+[36692]: https://github.com/rust-lang/rust/issues/36692
+[36430]: https://github.com/rust-lang/rust/issues/36430
+
+See the [detailed release notes][notes] for more.
+
+#### Cargo features
+
+As for Cargo, [RFC 1721] has been implemented. The motivation for this feature
+is that Cargo can now compile objects for statically linking against the msvcrt
+on the MSVC platform. But, meachnically speaking, this means that Cargo will
```
Comment (retep998) [inline _posts/2016-12-22-Rust-1.14.md]: mechanically

Comment (durka) [inline _posts/2016-12-22-Rust-1.14.md]: mechanically

File: _posts/2016-12-22-Rust-1.14.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+> channels, enabling you to easily switch between stable, beta, and nightly
+> compilers and keep them updated. It makes cross-compiling simpler with binary
+> builds of the standard library for common platforms. And it runs on all
+> platforms Rust supports, including Windows.
+
+[rustup]: https://internals.rust-lang.org/t/beta-testing-rustup-rs/3316/203
+
+We had [a previous post about Rustup][prev] back in May. You can learn more
+about it there, or by checking it out [on
+GitHub](https://github.com/rust-lang-nursery/rustup.rs)
```
Comment (xen0n) [inline _posts/2016-12-22-Rust-1.14.md]: Lacking a period at the end (possibly due to the inline link also containing one)

File: _posts/2016-12-22-Rust-1.14.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+website, and check out the [detailed release notes for 1.14.0][notes] on GitHub.
+1230 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1140-2016-12-22
+
+### What's in 1.14.0 stable
+
+One of the biggest features in Rust 1.14 isn't actually in Rust 1.14: the
+[rustup tool has reached a 1.0 release][rustup], and is now the recomended way
```
Comment (xen0n) [inline _posts/2016-12-22-Rust-1.14.md]: recommended

File: _posts/2016-12-22-Rust-1.14.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+`rustup target add`.
+
+These platforms are all 'tier 2', please see our page on [platform support] for
+more details.
+
+[platform support]: https://forge.rust-lang.org/platform-support.html
+
+The landing of MIR over the last few releases means that a [number of
+improvements to compile times] have landed, with more coming in the future.
+Exact numbers are hard, as they depend on what code you're compiling, but
```
Comment (aturon) [inline _posts/2016-12-22-Rust-1.14.md]: The "exact numbers are hard" seems disingenuous; we have a benchmark suite. If we don't have good overall numbers to report, I'd just drop this sentence.

File: _posts/2016-12-22-Rust-1.14.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+This was a inconsistency, and so with RFC 1492 stabilized, compiles fine as of
+this release. This applies to more situations than tuples; please see [the
+RFC][RFC 1492] for more details.
+
+[RFC 1492]: https://github.com/rust-lang/rfcs/blob/master/text/1492-dotdot-in-patterns.md
+
+#### Library stabilizations
+
+There have been a number of additions to the standard library, but they don't
+fit into particularly nice categories for this release. Here's the highlights:
```
Comment (aturon) [inline _posts/2016-12-22-Rust-1.14.md]: It'd be great to provide a sentence description of each of the listed changes.

File: _posts/2016-12-22-Rust-1.14.md [hunk 11]
```diff
@@ -0,0 +1,356 @@
...
+[37094]: https://github.com/rust-lang/rust/issues/37094
+[37470]: https://github.com/rust-lang/rust/issues/37470
+[36692]: https://github.com/rust-lang/rust/issues/36692
+[36430]: https://github.com/rust-lang/rust/issues/36430
+
+See the [detailed release notes][notes] for more.
+
+#### Cargo features
+
+As for Cargo, [RFC 1721] has been implemented. The motivation for this feature
```
Comment (aturon) [inline _posts/2016-12-22-Rust-1.14.md]: You say "the motivation for this feature" without actually saying what the feature is.

File: _posts/2016-12-22-Rust-1.14.md [hunk 12]
```diff
@@ -0,0 +1,356 @@
...
+[Jan-Erik's slides] for the workshop he ran at [Rust Belt Rust] for some
+examples, or check out [Tim's example of the classic TodoMVC project][todomvc].
+This implementation builds off of his [webplatform
+crate](https://crates.io/crates/webplatform), which exposes the DOM to Rust.
+
+[Jan-Erik's slides]: http://www.hellorust.com/emscripten/
+[Rust Belt Rust]: http://www.rust-belt-rust.com/sessions/
+[todomvc]: http://timryan.org/rust-todomvc/
+
+Speaking of platforms, a large number of platforms have gained additional
```
Comment (erickt) [inline _posts/2016-12-22-Rust-1.14.md]: If we're talking about experimental support for new platforms, how about mentioning [xargo](https://github.com/japaric/xargo)?

File: _posts/2016-12-22-Rust-1.14.md (hunk 11 above)
Comment (steveklabnik) [inline _posts/2016-12-22-Rust-1.14.md]: I was considering "RFC 1721" as the feature, and then explaining the details after, but maybe that's too convoluted. Any suggetsions?

**Comment by steveklabnik** [comment]
I thought I had my spellcheck on, sorry about that everyone. Should have everything done except @aturon 's two comments about sentences for library changes, as well as the slight weirdness of phrasing. I might knock those out in the morning.

**Comment by brson** [comment]
I pushed some minor typo fixes.

**Comment by steveklabnik** [comment]
Okay, this should be good to go 👍

**Pull request merged**